vehicle/toyota/identity.go

```
@@ -0,0 +1,183 @@
+		if err = v.DoJSON(req, &token); err != nil {
+			return nil, err
+		}
+		return &token, nil
+	}
+
+	// Otherwise continue with Auth flow
+	var res Auth
+	if err = v.DoJSON(req, &res); err != nil {
+		return nil, err
+	}
+
+	// Continue authentication flow
+	return v.authenticate(res, user, password, passwordSet)
+}
+
+func (v *Identity) authorize(token Token) (string, error) {
+	uri := fmt.Sprintf("%s/%s", BaseUrl, AuthorizationPath)
+	req, err := request.New(http.MethodGet, uri, nil, map[string]string{
+		"Cookie": fmt.Sprintf("iPlanetDirectoryPro=%s", token.TokenID),
+	})
+	if err != nil {
+		return "", err
+	}
+	var param request.InterceptResult
+	v.Client.CheckRedirect, param = request.InterceptRedirect("code", true)
+	var code string
+	if _, err = v.Do(req); err == nil {
+		code, err = param()
+	}
+	return code, err
+}
+
+func (v *Identity) fetchTokenCredentials(code string) error {
+	uri := fmt.Sprintf("%s/%s", BaseUrl, AccessTokenPath)
+
+	data := url.Values{
+		"client_id":     {ClientID},
+		"code":          {code},
+		"redirect_uri":  {RedirectURI},
+		"grant_type":    {"authorization_code"},
+		"code_verifier": {"plain"},
+	}
+
+	req, err := request.New(http.MethodPost, uri, strings.NewReader(data.Encode()), map[string]string{
+		"Authorization": "Basic b25lYXBwOm9uZWFwcA==",
+		"Content-Type":  "application/x-www-form-urlencoded",
+	})
+	if err != nil {
+		return err
+	}
+
+	var resp struct {
+		oauth2.Token
+		IDToken string `json:"id_token"`
+	}
+	if err = v.DoJSON(req, &resp); err != nil {
+		return fmt.Errorf("failed to fetch token credentials: %w", err)
+	}
+
+	// Parse ID token without verification to extract UUID
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+	token, _, err := parser.ParseUnverified(resp.IDToken, jwt.MapClaims{})
+	if err != nil {
+		return fmt.Errorf("failed to parse id token: %w", err)
+	}
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+		if uuid, ok := claims["uuid"].(string); ok {
+			v.uuid = uuid
```

Was passiert wenn das nicht gefunden wird?

---

Dann isses kaputt!

Im ernst: Habe error handling und ein fail fast eingeführt, falls der uuid claim nicht gefunden wird.

---

Sag ich doch- vor dem letzten Commit gabs hier kein Error Handling ;)